lib/normalize.js

```diff
@@ -1149,48 +1275,193 @@ const prepareBorderShorthands = (properties) => {
+  if (properties.has(BORDER_IMAGE)) {
+    properties.delete(BORDER_IMAGE);
+    properties.set(BORDER_IMAGE, imageItem);
   }
   return properties;
 };
 
-exports.prepareProperties = (properties, opt = {}) => {
-  const { globalObject } = opt;
-  const { positions } = borderElements;
+/**
+ * Processes shorthand properties from the shorthands map.
+ *
+ * @param {Map} shorthands - The map containing shorthand property groups.
+ * @returns {Map} A map of processed shorthand properties.
+ */
+const processShorthandProperties = (shorthands) => {
+  const shorthandItems = new Map();
+  for (const [property, item] of shorthands) {
+    const shorthandItem = shorthandProperties.get(property);
+    if (item.size === shorthandItem.shorthandFor.size && shorthandItem.position) {
+      const positionValues = [];
+      let priority = "";
+      for (const { value: longhandValue, priority: longhandPriority } of item.values()) {
+        positionValues.push(longhandValue);
+        if (longhandPriority) {
+          priority = longhandPriority;
+        }
+      }
+      const value = getPositionValue(positionValues, shorthandItem.position);
+      shorthandItems.set(property, createPropertyItem(property, value, priority));
+    }
+  }
+  return shorthandItems;
+};
+
+/**
+ * Processes border properties from the borders map, expanding and normalizing them.
+ *
+ * @param {Map} borders - The map containing accumulated border properties.
+ * @param {Object} parseOpt - Options for parsing values.
+ * @returns {Map} A map of fully processed and normalized border properties.
+ */
+const processBorderProperties = (borders, parseOpt) => {
+  const longhandProperties = new Map();
+  for (const [property, item] of borders) {
+    if (shorthandProperties.has(property)) {
+      const { value, priority } = item;
+      if (property === BORDER) {
+        const lineItems = border.parse(value, parseOpt);
+        for (const [key, initialValue] of border.initialValues) {
+          if (!Object.hasOwn(lineItems, key)) {
+            lineItems[key] = initialValue;
+          }
+        }
+        for (const lineProperty of Object.keys(lineItems)) {
+          let namePart, linePart;
+          const hyphenIndex = lineProperty.indexOf("-");
+          if (hyphenIndex !== -1) {
+            namePart = lineProperty.substring(0, hyphenIndex);
+            linePart = lineProperty.substring(hyphenIndex + 1);
+          } else {
+            // fallback for safety, though lineProperty from border.parse keys
+            // should have hyphen
+            namePart = lineProperty;
+            linePart = "";
+          }
+          const lineValue = lineItems[lineProperty];
+          for (const position of borderPositions) {
+            const longhandProperty = `${namePart}-${position}-${linePart}`;
+            const longhandItem = createPropertyItem(longhandProperty, lineValue, priority);
+            if (longhandProperties.has(longhandProperty)) {
```

Comment:
This pattern (lines 1346-1354) appears multiple times. Can we extract it into a helper?

Reply:
https://github.com/jsdom/cssstyle/pull/261/commits/6507ee383221a740a2861604b816180b0aef0eca

Please take a look.

Reply:
https://github.com/jsdom/cssstyle/pull/261/commits/15cd48cfdee3d26b21a9b65e1f6fe847cd9046a2
Is this what you're saying?